Implement AI Lab extension installation test, add playwright framework and tests structure

ScrewTSW opened the pull request:
### What does this PR do?
* Adds playwright testing framework and folder structure
* Implements basic AI Lab extension installation test

### Screenshot / video of UI
[podman-desktop-ai-lab-extension-installation.webm](https://github.com/containers/podman-desktop-extension-ai-lab/assets/16451875/a5debe24-5855-43cc-b9ec-eb4bef580c92)

<!-- If this PR is changing UI, please include 
screenshots or screencasts showing the difference -->

### What issues does this PR fix or reference?
containers/podman-desktop-internal/issues/174

<!-- Include any related issues from Podman Desktop 
repository (or from another issue tracker). -->

### How to test this PR?
* clone podman-desktop repository
* build podman in development mode:
    * yarn install --frozen-lockfile
    * yarn test:e2e:build
* return back to ai-studio repository
* yarn install --frozen-lockfile
* yarn test:e2e
<!-- Please explain steps to reproduce -->

### Additional context
* Currently the AI Lab extension is rendered inside a `<webview>` element, making it very difficult if not impossible to interact with it through DOM
* Current implementation is as far as I'm able to go with the playwright framework (it's possible I'm unaware of a workaround)

odockal reviewed (commented):
Looks pretty much alright, address the PR check complains.

ScrewTSW commented:
> Looks pretty much alright, address the PR check complains.

fixed the linter issues
![Code_WSQj1Emb5y](https://github.com/containers/podman-desktop-extension-ai-lab/assets/16451875/40b6a8e4-5149-47f1-b952-8a07a386061e)

odockal commented:
LGTM. But it would be nice to got someone to check yarn.lock since there is quite a lot of changes.

benoitf commented on .eslintrc.json:
```diff
@@ -19,7 +19,8 @@
     "project": [
-      "packages/*/tsconfig.json"
+      "packages/*/tsconfig.json",
+      "tests/playwright/tsconfig.json"
```
```suggestion
      "tests/*/tsconfig.json"
```

benoitf commented on tests/playwright/package.json:
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "ailab-tests-playwright",
```
```suggestion
  "name": "ai-lab-tests-playwright",
```

benoitf commented on tests/playwright/package.json:
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "ailab-tests-playwright",
+  "version": "0.0.1",
```
```suggestion
  "version": "0.4.0-next",
```

benoitf commented on tests/playwright/src/ai-lab-extension-installation.spec.ts:
```diff
@@ -0,0 +1,88 @@
+beforeAll(async () => {
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
+  pdRunner.setVideoAndTraceName('podman-desktop-ai-lab-extension-installation');
```
```suggestion
  pdRunner.setVideoAndTraceName('podman-desktop-extension-ai-lab-installation');
```
(to mimic repository name)

benoitf reviewed (commented):
it's missing lines in 
https://github.com/containers/podman-desktop-extension-ai-lab/blob/main/.github/workflows/release.yaml#L59-L61

to update version of package.json

ScrewTSW commented:
> it's missing lines in https://github.com/containers/podman-desktop-extension-ai-lab/blob/main/.github/workflows/release.yaml#L59-L61
> 
> to update version of package.json

implemented suggestion, added version fix

benoitf commented:
> Currently the AI Lab extension is rendered inside a <webview> element, making it very difficult if not impossible to interact with it through DOM

did you do the focus ?
@ScrewTSW https://github.com/microsoft/playwright/issues/9729#issuecomment-1646860598

benoitf commented:
@ScrewTSW next step is to add a workflow to launch the tests on each PR ?

ScrewTSW commented:
> > Currently the AI Lab extension is rendered inside a  element, making it very difficult if not impossible to interact with it through DOM
> 
> did you do the focus ? @ScrewTSW [microsoft/playwright#9729 (comment)](https://github.com/microsoft/playwright/issues/9729#issuecomment-1646860598)

Interesting workaround, thank you for the suggestion
